cmd/tsbs_generate_queries/uses/devops/common.go

```diff
@@ -157,41 +161,45 @@ func GetMaxAllLabel(dbName string, nHosts int) string {
 	return fmt.Sprintf("%s max of all CPU metrics, random %4d hosts, random %s by 1h", dbName, nHosts, MaxAllDuration)
 }
 
-func getRandomHosts(scale, nHosts int) []string {
-	if nHosts < 1 {
-		fatal("number of hosts cannot be < 1; got %d", nHosts)
+// getRandomHosts returns a subset of numHosts hostnames of a permutation of hostnames,
+// numbered from 0 to totalHostsScale.
+// Ex.: host_12, host_7, host_22 for numHosts=3 and totlaHostsScale=30
+func getRandomHosts(numHosts int, totalHostsScale int) []string {
```

Comment:
Changing the order of the arguments in this file worries me -- are you sure you got all the places where it may have changed?

Reply:
1. Regarding changing order of the arguments - main idea is that it seems to be more human-friendly to say `get 5 random items out of 12` - and I personally stumbled several times during code digging in this. Totally agree that  this is subjective, but still would be nice (from my point of view) to keep it
2. As to `all the places` - yes, hope to find and modify all the places - there are not so many of them and GUI helps a lot with this.

Reply:
Cool, I agree with 1. So if all the spots are gotten (since it is unexported it should only be in that package), then it's a fine change to me.

Reply:
Actually if it's not too much trouble, can you separate this out for a separate PR like you did for the other change?

Reply:
Yes, think this `getRandomHosts` refactoring can be extracted into separate PR. We may complete it before this big `ClickHouse` PR

Reply:
Extracted random refactoring into separate PR [here](https://github.com/timescale/tsbs/pull/27)
Let's move on with this new PR (random functions) and then return back to ClickHouse